### PR TITLE
Added update metadata methods

### PR DESCRIPTION
**Pull request by icellan**

### CODE REVIEW

File: src/transports/graphql.ts [hunk 1]
```diff
@@ -336,6 +360,96 @@ class TransportGraphQL implements TransportService {
...
+    `;
+    const variables = {
+      id,
+      metadata,
+    };
+
+    return this.doGraphQLMutation(query, variables, 'destination_metadata');
+  }
+
+  // UpdateDestinationMetadataByID updates the destination metadata by id
```
Comment (mrz1836) [inline src/transports/graphql.ts:393]: Comment is wrong function name

File: src/transports/graphql.ts [hunk 2]
```diff
@@ -455,6 +569,38 @@ class TransportGraphQL implements TransportService {
     return this.doGraphQLMutation(query, variables, 'transaction');
   }
 
+  async UpdateTransactionMetadata(txID: string, metadata: Metadata): Promise<Transaction> {
+    const query = gql`
+      mutation ($tx_id: String!, $metadata: Metadata!) {
+        transaction_metadata (
+        tx_id: $tx_id
```
Comment (mrz1836) [inline src/transports/graphql.ts:576]: I think it should be `id` vs `tx_id`

**Review by mrz1836 — CHANGES_REQUESTED** [review]
See comments @icellan

File: src/transports/graphql.ts (hunk 2 above)
Comment (icellan) [reply src/transports/graphql.ts:576]: @mrz1836 In graphql it is actually `tx_id`. We could change this, but that means a breaking change between clients and server

Comment (icellan) [reply src/transports/graphql.ts:576]: Better to change it, it should indeed be `id`. Will update all repos

**Comment by icellan** [comment]
@mrz1836 Bux-console needs the latest version of js-buxclient to interact with bux servers with the update.

**Pull request merged**